File: microservices/python/src/message_delivery_service.py
import logging

import requests

from src.config import CONFIG

logger = logging.getLogger("python-api")


class MessageDeliveryService:
    @staticmethod
    def send_message(message: dict) -> bool:

        if message is None:
            return True

        make_service_call = "Python" not in message["message"]

        if not make_service_call:
            logger.info("Service already published a message")
            return True

        message["message"] += "Python, "

        logger.info(f"Send message {message['message']}")
        logger.info(f"Send request to {CONFIG['external_api_path']}")

        result = requests.post(CONFIG["external_api_path"], json=message)

        return result.ok

File: microservices/rust/src/services/message_delivery_service.rs
use std::collections::HashMap;

use log::info;
use crate::config::Config;

use crate::models::message::Message;

pub async fn send_message(message: Message) -> bool {
    let make_service_call = !message.message.contains("Rust");

    if !make_service_call {
        info!("Service already published a message");
        return true;
    }

    let mut request = HashMap::new();
    request.insert("message", message.message + "Rust");

    let external_path = Config::external_api_path();

    info!("Send message {}", request.get("message").unwrap());
    info!("Send request to {}", external_path);

    let client = reqwest::Client::new();

    if !Config::need_to_publish() {
        info!("No need to publish request");
        return true;
    }

    let response = client.post(external_path)
        .json(&request)
        .send()
        .await;

    return match response {
        Ok(r) => true,
        Err(e) => false,
    };
}
